aks/README.md

```diff
@@ -0,0 +1,107 @@
+
+## Prerequisites
+1. Create a resource group if you haven't got one ready for use. 
+```bash
+az deployment create \
+  --name MY_RESOURCE_GROUP_NAME \
+  --location eastus \
+  --template-file ./resource-group.json
+```
+
+az deployment create \
+  --name joshua-rg8 \
+  --location eastus \
+  --template-file ./resource-group.json
+
+2. Don't create more than one AKS cluster in the same subnet.
+3. AKS clusters may **not** use _169.254.0.0/16, 172.30.0.0/16, 172.31.0.0/16, or 192.0.2.0/24_ for the Kubernetes service address range.
+4. To interact with Azure APIs, an AKS cluster requires an Azure Active Directory (AD) [Service Principal](https://docs.microsoft.com/en-us/azure/aks/kubernetes-service-principal). The Service Principal (SP) is needed to dynamically create and manage other Azure resources such as an Azure load balancer, container registry (ACR) etc
+
+Unless specified, the SP is assigned the 'Contributor' role by default - this has full permissions to read and write to your Azure account which isn't ideal from a security standpoint.
+Instead, create a new SP without any roles assigned to it. The template will then append the necessary permissions.
+ 
+Create an SP like so:
+```bash
+az ad sp create-for-rbac --name "MY_RBAC_NAME" --skip-assignment
+# Get the objectId using the appId from the response
+az ad sp show --id <appId> --query objectId
+```
+
+NOTE: When you create a Service Principal, take note of the following in the response (redacted eg shown below):
+```bash
+{
+  "appId": "2d3.....748",
+  "displayName": "abcd..",
+  "name": "http://abcd..",
+  "password": "a07.....c0d",
+  "tenant": "172.....f81"
+}
+```
+- **Service principal Client ID** is your _**appId**_
+- **Service principal Client Secret** is the _**password**_ value
+- **Service principal Tenant ID** is the _**tenant**_ value
+
+
+For this deployment we will provision (to your SP) and use the following permissions: 
+```bash
+# permissions on the subnet within your virtual network i.e Azure's `Network Contributor` role
+- Microsoft.Network/virtualNetworks/subnets/join/action
+- Microsoft.Network/virtualNetworks/subnets/read
+
+# The VM where Helm is installed requires permissions to communicate with the cluster and provision Besu i.e Azure's `Azure Kubernetes Service Cluster Admin Role` role
+- Microsoft.ContainerService/managedClusters/listClusterAdminCredential/action
+```
+
+## Description
+This deployment template will create an AKS cluster for you in Azure, as well as a VM to run helm from to provision the cluster. Part of the process is that it will install Helm3 and the [Besu Helm charts](https://github.com/PegaSysEng/besu-kubernetes) in the home directory of the VM.
+
```

Comment:
Please add some instructions on how to actually do the deployment

Reply:
done